### PR TITLE
eth-utils - v5.1.0 :snowflake: 

### PR DESCRIPTION
**Pull request by Andrii-Osipov**
eth-utils 5.1.0 :snowflake:

**Destination channel:** Snowflake

- [PKG-5927](https://anaconda.atlassian.net/browse/PKG-5927) 
- [Upstream repository](https://github.com/ethereum/eth-utils/tree/v5.1.0)

[PKG-5927]: https://anaconda.atlassian.net/browse/PKG-5927?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,49 +1,69 @@
...
+    - eth-hash >=0.3.1
+    - eth-typing >=5.0.0
+    - toolz >0.8.2
+    - cytoolz >=0.10.1
+  run_constrained:
+    - tox >=4.0.0
+    - pre-commit >=3.4.0
+    - build >=0.9.0
+    - bump_my_version >=0.19.0
+    - pycryptodome >=3.6.6,<4
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
```
These are all development dependencies.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,49 +1,69 @@
...
-    - cytoolz <1.0.0,>=0.10.1  # [python_impl == 'cpython']
-    - eth-hash <1.0.0,>=0.3.1
-    - eth-typing <4.0.0,>=3.0.0
     - python
-    - toolz <1,>0.8.2  # [python_impl == 'pypy']
-    - cached-property <2,>=1.5.2  # [py<3.8]
+    - eth-hash >=0.3.1
+    - eth-typing >=5.0.0
+    - toolz >0.8.2
+    - cytoolz >=0.10.1
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
    - toolz >0.8.2  # [python_impl == 'pypy']
    - cytoolz >=0.10.1  # [python_impl == 'cpython']
```
If `python_impl` is not available, then we could either make sure to add it to aggregate's cbc file or just remove `toolz`.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,49 +1,69 @@
...
     - eth_utils.curried
+  commands:
+    - pip check
+    - pytest -vv tests
   requires:
     - pip
+    - hypothesis >=4.43.0
+    - pytest >=7.0.0
+    - pytest-xdist >=2.4.0
+    - mypy ==1.10.0
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:49]:
```suggestion
```
Both are unused.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,49 +1,69 @@
...
 
 about:
   home: https://github.com/ethereum/eth_utils
   summary: Common utility functions for ethereum codebases.
   license: MIT
+  license_family: MIT
   license_file: LICENSE
+  dev_url: https://github.com/ethereum/eth-utils
+  doc_url: https://eth-utils.readthedocs.io/en/stable/
+  description: Common utility functions for ethereum codebases.
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
  description: Common utility functions for python code that interacts with Ethereum
```

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,49 +1,69 @@
...
   requires:
     - pip
+    - hypothesis >=4.43.0
+    - pytest >=7.0.0
+    - pytest-xdist >=2.4.0
+    - mypy ==1.10.0
 
 about:
   home: https://github.com/ethereum/eth_utils
   summary: Common utility functions for ethereum codebases.
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
  summary: Utility functions for working with ethereum related codebases.
```

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:49]: Yeah, it looks like there are no `mypy` imports anywhere https://github.com/search?q=repo%3Aethereum%2Feth-utils%20mypy&type=code.

Comment (Andrii-Osipov) [reply recipe/meta.yaml:49]: https://github.com/ethereum/eth-utils/blob/main/setup.py#L27
Needs for the pytest

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:49]: mypy is only ran from their pre-commit configs: https://github.com/ethereum/eth-utils/blob/main/.pre-commit-config.yaml#L46-L53.

We never need mypy.

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:49]: Ph wait, https://github.com/ethereum/eth-utils/blob/main/tests/core/functional-utils/test_type_inference.py#L6-L8.

Anyway, `pytest-xdist` is not needed.

Comment (boldorider4) [reply recipe/meta.yaml:49]: `pypy` is merely a test dependency here so it goes into the `test: requires:` section.

**Review by danpetry — APPROVED** [review]

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,49 +1,63 @@
...
-    - cytoolz <1.0.0,>=0.10.1  # [python_impl == 'cpython']
-    - eth-hash <1.0.0,>=0.3.1
-    - eth-typing <4.0.0,>=3.0.0
     - python
-    - toolz <1,>0.8.2  # [python_impl == 'pypy']
-    - cached-property <2,>=1.5.2  # [py<3.8]
+    - eth-hash >=0.3.1
+    - eth-typing >=5.0.0
+    - toolz >0.8.2 # [python_impl == 'pypy']
+    - cytoolz >=0.10.1 # [python_impl == 'cpython']
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - cytoolz >=0.10.1
```
As we do not build for pypy

**Review by skupr-anaconda — APPROVED** [review]
LGTM but removing unneeded selectors

**Pull request merged**